index.js

```diff
@@ -71,12 +43,30 @@ LocaleProcessor.prototype.transform = function (localeName, fields, pluralFn) {
 LocaleProcessor.prototype.processString = function (inputString, filename) {
     var localeName = path.basename(filename, path.extname(filename));
 
-    var cldr = process(localeName, {
-        plurals: true,
-        fields:  ['year', 'month', 'day', 'hour', 'minute', 'second']
+    localeName = cldrLocales.locales.normalize(localeName);
+
+    var out   = {};
+    var parts = localeName.split('-');
+    
+    var data = cldr({
+        locales:        [localeName],
+        pluralRules:    true,
+        relativeFields: ['year', 'month', 'day', 'hour', 'minute', 'second']
+    });
+
+    parts.map(function(part, idx) {
+        /* 
+         * if the locale is 'zh-Hant-TW' this map will build an
+         * array of strings ['zh', 'zh-Hant', 'zh-Hant-TW']
+         * which is used to walk the CLDR results to progressively
+         * build up the results object
+         */
```

Comment:
I found it wasn't working, at least as expected.  For example, if I am targeting `zh-Hant-TW` then the object I get back from cldr function above is an object containing `zh` and `zh-Hant` and `zh-Hant` was not an aggregating of `zh` and `zh-Hant`.  Which is why I had to walk the results to build it up..